nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/PutCassandraRecord.java

```diff
@@ -394,11 +394,17 @@ private Long convertFieldObjectToLong(String name, Object value) {
 
             // Check that if the update method is set to increment or decrement that the batch statement type is set to
             // unlogged or counter (or USE_ATTR_TYPE, which we cannot check at this point).
+            String updateMethod = validationContext.getProperty(UPDATE_METHOD).getValue();
             String batchStatementType = validationContext.getProperty(BATCH_STATEMENT_TYPE).getValue();
-            if (!Sets.newHashSet(COUNTER_TYPE.getValue(), UNLOGGED_TYPE.getValue(), BATCH_STATEMENT_TYPE_USE_ATTR_TYPE.getValue()).contains(batchStatementType)) {
```

Comment:
Removing this causes a CheckStyle error for the unused import for Sets. I can remove that while merging, but if I end up leaving another comment asking for changes please change this too and run Maven with `-Pcontrib-check` for the cassandra bundle to make sure the build will succeed, thanks!